services/rpcfilters/latest_logs.go

```diff
@@ -18,16 +18,13 @@ type ContextCaller interface {
 }
 
 func pollLogs(client ContextCaller, f *logsFilter, timeout, period time.Duration) {
-	adjusted := false
 	query := func() {
 		ctx, cancel := context.WithTimeout(f.ctx, timeout)
-		logs, err := getLogs(ctx, client, f.crit)
-		cancel()
+		defer cancel()
+		logs, err := getLogs(ctx, client, f.criteria())
 		if err != nil {
-			log.Error("failed to get logs", "criteria", f.crit, "error", err)
-		} else if !adjusted {
-			adjustFromBlock(&f.crit)
-			adjusted = true
+			log.Error("error fetch logs", "criteria", f.crit, "error", err)
```

Comment:
lowercasing error message is needed for chaining errors, but here we're printing it to console, so:
```suggestion
			log.Error("Error fetching logs", "criteria", f.crit, "error", err)
```

Reply:
thanks, didnt notice it